File: corporate-reporting/src/types/DirectorateUpdate.ts
import { IDirectorate } from "./Directorate";
import { IRagOption } from "./RagOption";
import { IReportingProgressUpdate, ReportingProgressUpdate } from "./ReportingProgressUpdate";

export interface IDirectorateUpdate extends IReportingProgressUpdate {
    DirectorateID: number;
    OverallRagOptionID: number;
    FinanceRagOptionID: number;
    FinanceComment: string;
    PeopleRagOptionID: number;
    PeopleComment: string;
    MilestonesRagOptionID: number;
    MilestonesComment: string;
    MetricsRagOptionID: number;
    MetricsComment: string;
    ProgressUpdate: string;
    FutureActions: string;
    Escalations: string;
    Directorate?: IDirectorate;
    OverallRagOption?: IRagOption;
    FinanceRagOption?: IRagOption;
    PeopleRagOption?: IRagOption;
    MilestonesRagOption?: IRagOption;
    MetricsRagOption?: IRagOption;
}

export class DirectorateUpdate extends ReportingProgressUpdate implements IDirectorateUpdate {
    public DirectorateID = null;
    public OverallRagOptionID = null;
    public FinanceRagOptionID = null;
    public FinanceComment = '';
    public PeopleRagOptionID = null;
    public PeopleComment = '';
    public MilestonesRagOptionID = null;
    public MilestonesComment = '';
    public MetricsRagOptionID = null;
    public MetricsComment = '';
    public ProgressUpdate = '';
    public FutureActions = '';
    public Escalations = '';

    public Directorate = null;
    public UpdateUser = null;

    constructor(directorateId?: number, period?: Date) {
        super(period);
        this.DirectorateID = directorateId;
    }
}

File: ORB.Core/Models/DirectorateUpdate.cs
using System;
using System.Collections.Generic;

namespace ORB.Core.Models
{
    public partial class DirectorateUpdate : EntityUpdate
    {
        public string Comment { get; set; }
        public int? RagOptionID { get; set; }
        public int? DirectorateID { get; set; }
        public int? OverallRagOptionID { get; set; }
        public int? FinanceRagOptionID { get; set; }
        public string FinanceComment { get; set; }
        public int? PeopleRagOptionID { get; set; }
        public string PeopleComment { get; set; }
        public int? MilestonesRagOptionID { get; set; }
        public string MilestonesComment { get; set; }
        public int? MetricsRagOptionID { get; set; }
        public string MetricsComment { get; set; }
        public string ProgressUpdate { get; set; }
        public string FutureActions { get; set; }
        public string Escalations { get; set; }
        public bool? ToBeClosed { get; set; }
        public DateTime? UpdatePeriod { get; set; }

        public virtual Directorate Directorate { get; set; }
        public virtual RagOption FinanceRagOption { get; set; }
        public virtual RagOption MetricsRagOption { get; set; }
        public virtual RagOption MilestonesRagOption { get; set; }
        public virtual RagOption OverallRagOption { get; set; }
        public virtual RagOption PeopleRagOption { get; set; }
    }
}
